icu4c/source/common/cmemory.h

```diff
@@ -725,6 +725,12 @@ class MemoryPool : public UMemory {
     }
 
     MemoryPool& operator=(MemoryPool&& other) U_NOEXCEPT {
+        // Since `this` may contain instances that need to be deleted, we can't
```

Comment:
Ah, this is an important change, the original assumption here was that the object being assigned to would always be empty.